pkg/sql/opt/memo/statistics_builder.go

```diff
@@ -863,9 +864,20 @@ func (sb *statisticsBuilder) constrainScan(
 				// estimate 0 rows.)
 				s.RowCount = max(inputHist.ValuesCount(), 1)
 				if colStat, ok := s.ColStats.Lookup(colSet); ok {
-					colStat.Histogram = inputHist.InvertedFilter(scan.InvertedConstraint)
-					histCols.Add(invertedConstrainedCol)
-					sb.updateDistinctCountFromHistogram(colStat, inputStat.DistinctCount)
+					// If we have a histogram that doesn't contain DBytes, we know we're
+					// looking at a forward histogram, since inverted histograms always
+					// are of type DBytes. We cannot use the histogram if this is the
+					// case.
+					// Note that this is a sketchy situation (no pun intended), because
+					// if we one day supported inverted indexes on bytes types, this check
+					// wouldn't properly reject the histogram. The proper fix, tracked
+					// in #50655, is to add a type field onto the persisted histogram.
+					if inputHist.BucketCount() == 0 ||
+						inputHist.Bucket(0).UpperBound.ResolvedType().Family() == types.BytesFamily {
```

Comment:
Internally the type we use for inverted columns is `types.EncodedKey`: https://github.com/cockroachdb/cockroach/blob/10a054b63d9403aa48ef2f7d7b3615a938879240/pkg/sql/types/types.go#L479-L485

But I guess we write the type bytes to the persisted histogram?

Reply:
Yeah, we do - there is some commentary about it where we write the persisted inverted histograms.